lnrpc/rpc.proto

```diff
@@ -1579,14 +1595,82 @@ message OpenChannelRequest {
 
     /// Whether unconfirmed outputs should be used as inputs for the funding transaction.
     bool spend_unconfirmed = 12 [json_name = "spend_unconfirmed"];
+
+    /**
+    Funding shims are an optional argument that allow the caller to intercept
+    certain funding functionality. For example, a shim can be provided to use a
+    particular key for the commitment key (ideally cold) rather than use one
+    this is generated by the wallet as normal, or signal that signing will be
+    carried out in an interactive manner (PSBT based).
+    */
+    FundingShim funding_shim = 13;
```

Comment:
What was the original reason for all the `[json_name = "..."]` in this file? Are they still needed? If yes, then they are missing throughout the changes for this PR.

Reply:
They're needed for things that will be displayed by `lncli` on the command line. It's only for JSON encoding of the field names. By default, the proto library will default to using CamelCase, but we use snake_case for our JSON output on the command line, hence the need for specifying the name manually.